scene/resources/2d/tile_set.h

```diff
@@ -265,8 +265,8 @@ class TileSet : public Resource {
 	class TerrainsPattern {
 		bool valid = false;
 		int terrain = -1;
-		int bits[TileSet::CELL_NEIGHBOR_MAX];
-		bool is_valid_bit[TileSet::CELL_NEIGHBOR_MAX];
+		int bits[TileSet::CELL_NEIGHBOR_MAX] = {};
+		bool is_valid_bit[TileSet::CELL_NEIGHBOR_MAX] = {};
```

Comment:
Fixes this:
![image](https://github.com/godotengine/godot/assets/4701338/e19c6bf2-4e5c-4781-9eea-dfe5aa5f2451)

CC @groud